lib/serializer-utils.js

```diff
@@ -181,8 +181,10 @@ module.exports = function (collectionName, record, payload, opts) {
       if (opts.includedLinks) {
         included.links = getLinks(dest, opts.includedLinks);
       }
-
-      pushToIncluded(payload, included);
+      // only add to included array if the resource is defined
+      if(id !== 'undefined') {
+        pushToIncluded(payload, included);
+      }
     }
 
     return id !== 'undefined' ? { type: type, id: id } : null;
```

Comment:
Assuming you refactor the `getRef` function to return `undefined` and not `'undefined'`, this line become:
`return typeof id !== 'undefined' ? { type: type, id: id } : null;`